lib/bundler/feature_flag.rb

```diff
@@ -19,12 +19,21 @@ def self.settings_option(key, &default)
     def self.settings_method(name, key, &default)
       define_method(name) do
         value = Bundler.settings[key]
-        value = instance_eval(&default) if value.nil? && !default.nil?
+        value = instance_eval(&default) if value.nil?
```

Comment:
This is kind of unrelated to the PR, but I saw it while working on "env based feature settings". Currently we're never passing `default` as `nil`here, and I don't think it would make sense that we did that?

Reply:
I extracted this change to #6884 because it was unrelated, so it should be discussed separately.